src/data/value/mod.rs

```diff
@@ -170,6 +169,63 @@ impl Value {
         }
     }
 
+    pub fn cast(&self, data_type: &DataType) -> Result<Self> {
+        match (data_type, self) {
+            // Same as
+            (DataType::Boolean, Value::Bool(_))
+            | (DataType::Int, Value::I64(_))
+            | (DataType::Float(_), Value::F64(_))
+            | (DataType::Text, Value::Str(_)) => Ok(self.clone()),
+            // Boolean
+            (DataType::Boolean, Value::Str(value)) => Ok(match value.to_uppercase().as_str() {
+                "TRUE" => Ok(Value::Bool(true)),
+                "FALSE" => Ok(Value::Bool(false)),
+                _ => Err(ValueError::ImpossibleCast),
+            }?),
+            (DataType::Boolean, Value::I64(value)) => Ok(match value {
+                1 => Ok(Value::Bool(true)),
+                0 => Ok(Value::Bool(false)),
+                _ => Err(ValueError::ImpossibleCast),
+            }?),
+            (DataType::Boolean, Value::F64(value)) =>
+            {
+                #[allow(clippy::float_cmp)]
+                if *value == 1.0 {
+                    Ok(Value::Bool(true))
+                } else if *value == 0.0 {
+                    Ok(Value::Bool(false))
+                } else {
+                    Err(ValueError::ImpossibleCast.into())
+                }
```

Comment:
![image](https://user-images.githubusercontent.com/2025065/109625113-4647af00-7b82-11eb-963c-d84f3b6e6091.png)

This can make clippy satisfied.
Because `==` operator assumes that value implemented `Eq` but float only has `PartialEq`.
However, still `==` can be used for `PartialEq` but not preferred by strict clippy.

Reply:
Oh okay thanks